### DISCUSSION TITLE
[Lecture 12] Ask for understanding slide

### DISCUSSION THREAD

**Discussion by Akais24** [prosyslab-classroom/cs424-program-reasoning#436] [Q&A] [answered]
### Your Name

황재영

### Question

<img width="955" alt="image" src="https://github.com/user-attachments/assets/5affeab5-f760-4881-bef4-50acb2e6a65f">

HW4를 하기 위해 강의 자료를 훑던 중 Lecture 12 Slide 5에 대해서 이해가 되지 않는 부분이 있어서 질문드리려고 했습니다
Top-down + Eq. Reduction 를 하기 위해 if not equiv(p, p’) 가 추가되었는데,
p로부터 유래된 p’이 p와 동일할 수 있는 경우가 있는지 잘 모르겠습니다.
아니면 뭔가 동일함을 의미하는 것이 아닌, 다른 operation인지 궁금합니다
제 생각에는 Q에 존재하는 원소와 equiv할 것 같다고 느꼈습니다

감사합니다

---

While reviewing the lecture materials for HW4, I came across a part in Lecture 12 Slide 5 that I don't understand, so I wanted to ask a question.
For Top-down + Eq. Reduction, "if not equiv(p, p')" was added, but I'm not sure if there's a case where p' derived from p could be identical to p.
Or I'm wondering if it's not meant to indicate equality, but rather a different operation.
I felt that it might be equivalent to an element existing in Q.
Thank you.

**Comment by spearo2** [comment, marked as answer]
x 와 x + 0, x + x - x의 경우를 생각해보시면 될것 같습니다.

==================

You can consider cases like x, x + 0, and x + x - x.

**Reply by Akais24** [reply]
앗 그런데 제 생각에는 unroll을 한다고 생각하면
p = x + S인 경우에 
p' = [x + 0, x + 1, x + (S+S)...] 등과 같이 생각하였는데
그럼 이 경우에 걸러지는 것은 x + 0 뿐인 걸까요?
아니면 제가 unroll에 대한 로직을 잘못 이해하고 있는 걸까요?

---

Oh, but in my opinion, if we think about unrolling,
in the case where p = x + S,
I thought p' would be like [x + 0, x + 1, x + (S+S)...] and so on.
In this case, is x + 0 the only one that gets filtered out?
Or am I misunderstanding the logic of unrolling?

**Reply by spearo2** [reply]
그 부분을 고민하는것이 과제의 핵심입니다!
조교는 구현부분에서 도움을 줄 수 없다는점을 이해해 주시면 감사드리겠습니다.

=======================

That would be the main idea of the homework.
Please, understand that TAs cannot direct the implementation.

**Reply by KihongHeo** [reply]
여러가지 방법으로 할수 있겠지요. 강의록에는 한 가지만 간략히 소개한 것이고요. 정답은 없습니다. 창의력을 발휘해주세요.

---
There are various ways to do it. The lecture notes provide only one brief example. There’s no single correct answer, so please come up with your own idea.
